cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/ZookeeperScaler.java

```diff
@@ -0,0 +1,351 @@
+     * @return
+     */
+    private Future<Void> closeConnectionAsync() {
+        Promise<Void> closePromise = Promise.promise();
+
+        log.debug("Closing Zookeeper connection to {}", zookeeperConnectionString);
+
+        vertx.executeBlocking(promise -> {
+            closeConnectionSync();
+            promise.complete();
+        }, true, closePromise);
+
+        return closePromise.future();
+    }
+
+    /**
+     * Closes the Zookeeper connection synchronously
+     */
+    private void closeConnectionSync() {
+        if (zkAdmin != null)    {
+            try {
+                zkAdmin.close();
+            } catch (InterruptedException e) {
+                log.warn("Failed to close the ZooKeeperAdmin", e);
+            } finally {
+                zkAdmin = null;
+            }
+        }
+    }
+
+    /**
+     * Generates the TLS configuration for Zookeeper.
+     *
+     * @return
+     */
+    private ZKClientConfig getClientConfig()  {
+        ZKClientConfig clientConfig = new ZKClientConfig();
+
+        clientConfig.setProperty("zookeeper.clientCnxnSocket", "org.apache.zookeeper.ClientCnxnSocketNetty");
+        clientConfig.setProperty("zookeeper.client.secure", "true");
+        clientConfig.setProperty("zookeeper.ssl.trustStore.location", trustStoreFile.getAbsolutePath());
+        clientConfig.setProperty("zookeeper.ssl.trustStore.password", trustStorePassword);
+        clientConfig.setProperty("zookeeper.ssl.trustStore.type", "PKCS12");
+        clientConfig.setProperty("zookeeper.ssl.keyStore.location", keyStoreFile.getAbsolutePath());
+        clientConfig.setProperty("zookeeper.ssl.keyStore.password", keyStorePassword);
+        clientConfig.setProperty("zookeeper.ssl.keyStore.type", "PKCS12");
+
+        return clientConfig;
+    }
+
+    /**
+     * Converts the map with configuration to List of Strings which is the format in which the ZookeeperAdmin client
+     * expects the new configuration.
+     *
+     * @param servers   Map with Zookeeper configuration
+     * @return          List with Zookeeper configuration
+     */
+    /*test*/ static List<String> serversMapToList(Map<String, String> servers)  {
+        List<String> serversList = new ArrayList<String>(servers.size());
+
+        for (Map.Entry entry : servers.entrySet())  {
+            serversList.add(String.format("%s=%s", entry.getKey(), entry.getValue()));
+        }
+
+        return serversList;
+    }
+
+    /**
+     * Parse the byte array we get from Zookeeper into a map we use internally. During the parsing, only server
+     * configurations are used. The version field is ignored.
+     *
+     * @param byteConfig    byte[] from Zookeeper client
+     * @return              Map with Zookeeper configuration
+     */
+    /*test*/ static Map<String, String> parseConfig(byte[] byteConfig) {
+        String config = new String(byteConfig, StandardCharsets.US_ASCII);
+
+        Map<String, String> configMap = Util.parseMap(config);
```

Comment:
`Util.parseMap()` was originally written for our own 'image map' syntax. Are you sure it's the right thing to use for ZK admin responses?

Reply:
Well, it was there and does exactly what we need. What do you think would be better? 

Copying the method into the `ZookeeperScaler` class and keeping it as a duplicate? Probably the only different thing is the `,` parsing which we do not need for Zookeeper (and strictly speaking we do not use it in the image maps either AFAIK - at least not in the main YAML).

Reply:
I'm fine with using `Util.parseMap()` if it happens to be compatible with the ZK output. I was just checking.